sklearn/metrics/_classification_threshold.py

```diff
@@ -0,0 +1,122 @@
+from numbers import Integral
+
+import numpy as np
+
+from ._scorer import _BaseScorer
+
+
+def _threshold_scores_to_class_labels(y_score, threshold, classes, pos_label):
+    """Threshold `y_score` and return the associated class labels."""
+    if pos_label is None:
+        map_thresholded_score_to_label = np.array([0, 1])
+    else:
+        pos_label_idx = np.flatnonzero(classes == pos_label)[0]
+        neg_label_idx = np.flatnonzero(classes != pos_label)[0]
+        map_thresholded_score_to_label = np.array([neg_label_idx, pos_label_idx])
+
+    return classes[map_thresholded_score_to_label[(y_score >= threshold).astype(int)]]
+
+
+class _CurveScorer(_BaseScorer):
```

Comment:
I think that we should move those in the `_score.py` file. We already have the `_Scorer` there.

Reply:
I created this file because I thought that it made sense to have the new function to live here. But I'll move it.
Side question: do you think that `sklearn.metrics.decision_threshold_curve` should also be inside `_score.py`?